bpf/tests/lib/egressgw.h

```diff
@@ -209,47 +209,50 @@ static __always_inline int egressgw_snat_check(const struct __ctx_buff *ctx,
 			test_fatal("dst IP has changed");
 	}
 
-	/* Lookup the SNAT mapping for the original packet to determine the new source port */
-	struct ipv4_ct_tuple tuple = {
-		.daddr   = CLIENT_IP,
-		.saddr   = EXTERNAL_SVC_IP,
-		.dport   = EXTERNAL_SVC_PORT,
-		.sport   = client_port(test_ctx.test),
-		.nexthdr = IPPROTO_TCP,
-		.flags = TUPLE_F_OUT,
-	};
-	struct ct_entry *ct_entry = map_lookup_elem(get_ct_map4(&tuple), &tuple);
+	/* SNAT happens *after* redirect: */
+	if (!test_ctx.redirect) {
+		/* Lookup the SNAT mapping for the original packet to determine the new source port */
+		struct ipv4_ct_tuple tuple = {
+			.daddr   = CLIENT_IP,
+			.saddr   = EXTERNAL_SVC_IP,
+			.dport   = EXTERNAL_SVC_PORT,
+			.sport   = client_port(test_ctx.test),
+			.nexthdr = IPPROTO_TCP,
+			.flags = TUPLE_F_OUT,
+		};
+		struct ct_entry *ct_entry = map_lookup_elem(get_ct_map4(&tuple), &tuple);
 
-	if (!ct_entry)
-		test_fatal("no CT entry found");
+	  if (!ct_entry)
+	  	test_fatal("no CT entry found");
 
-	if (ct_entry->tx_packets != test_ctx.tx_packets)
-		test_fatal("bad TX packet count (expected %u, actual %u)",
-			   test_ctx.tx_packets, ct_entry->tx_packets)
-	if (ct_entry->rx_packets != test_ctx.rx_packets)
-		test_fatal("bad RX packet count (expected %u, actual %u)",
-			   test_ctx.rx_packets, ct_entry->rx_packets)
+	  if (ct_entry->tx_packets != test_ctx.tx_packets)
+	  	test_fatal("bad TX packet count (expected %u, actual %u)",
+	  		   test_ctx.tx_packets, ct_entry->tx_packets)
+	  if (ct_entry->rx_packets != test_ctx.rx_packets)
+	  	test_fatal("bad RX packet count (expected %u, actual %u)",
+	  		   test_ctx.rx_packets, ct_entry->rx_packets)
 
```

Comment:
This is where you hit the conflict, right? The resolution looks good (keep the old packet count checks), but the indentation looks slightly off.

Reply:
What do you mean slightly off? It's completely off! Fixed ;)